mobile_verifier/src/subscriber_mapping_activity.rs

```diff
@@ -191,43 +196,37 @@ where
 }
 
 async fn verify_known_carrier_key<AV>(
-    authorization_verifier: AV,
+    authorization_verifier: impl AsRef<AV>,
     public_key: &PublicKeyBinary,
 ) -> anyhow::Result<bool>
 where
     AV: AuthorizationVerifier,
 {
     authorization_verifier
+        .as_ref()
         .verify_authorized_key(public_key, NetworkKeyRole::MobileCarrier)
         .await
         .map_err(anyhow::Error::from)
 }
 
-async fn verify_subscriber_id<EV>(entity_verifier: EV, subscriber_id: &[u8]) -> anyhow::Result<bool>
+async fn verify_subscriber_id<EV>(
+    entity_verifier: impl AsRef<EV>,
+    subscriber_id: &[u8],
+) -> anyhow::Result<bool>
 where
     EV: EntityVerifier,
 {
     entity_verifier
+        .as_ref()
         .verify_rewardable_entity(subscriber_id)
         .await
         .map_err(anyhow::Error::from)
 }
 
-async fn is_valid(
-    activity: SubscriberMappingActivity,
-    status: SubscriberReportVerificationStatus,
-) -> anyhow::Result<Option<SubscriberMappingActivity>> {
-    if status == SubscriberReportVerificationStatus::Valid {
-        Ok(Some(activity))
-    } else {
-        Ok(None)
-    }
-}
-
 impl<AV, EV> ManagedTask for SubscriberMappingActivityDaemon<AV, EV>
 where
-    AV: AuthorizationVerifier + Clone,
-    EV: EntityVerifier + Clone + Send + Sync + 'static,
+    AV: AuthorizationVerifier,
+    EV: EntityVerifier + Send + Sync + 'static,
```

Comment:
Do we want to move `Send + Sync + 'static` to the definition of `EntityVerifier` as it will be required anywhere we use it?

Reply:
probably more concise: `EntityVerifier: Send + Sync + 'static { trait functions }`